### DISCUSSION TITLE
Huge cache files and Sudden high CPU usage

### DISCUSSION THREAD

**Discussion by cfftony** [vercel/next.js#57248] [Help]
### Summary

Next.js version: 13.5.6
App Router
Self hosting, Load Balancing,  4 nextjs instance (4 server Ubuntu  22.04 64)

**Sudden high CPU usage causing slow server response, resolved by restarting the Node.js service.
strace results:**
<img width="615" alt="image" src="https://github.com/vercel/next.js/assets/32441601/3f017fde-cd95-4d87-8134-f96163242a00">

**There are lots of errors in log:**
Failed to update prerender cache for 66332f43ff1efe6a09f7b7366e4a2b7793cfe640a1986e91f8bf7c4e475d5a 
errno: -28,                                                                                      
code: 'ENOSPC',                                                                                   
syscall: 'open',                                                                                
path: '/data/web/.next/cache/fetch-cache/66332f43ff1efe6a09f7b7366e4a2b7793cfe640a19 

**The '.next/cache/fetch-cache' folder currently has over 6.5 million records and is continuously growing.**

**df -i results:**
<img width="491" alt="image" src="https://github.com/vercel/next.js/assets/32441601/c33a9b50-0d2f-43e4-984d-a6f3a101d833">

**df -lh results:**
<img width="422" alt="image" src="https://github.com/vercel/next.js/assets/32441601/b3a59e81-5ed8-4b81-abd9-fc7490e6c1f6">

**ulimit -a results:**
<img width="462" alt="image" src="https://github.com/vercel/next.js/assets/32441601/e50310bd-9757-4fee-8346-42f8ea2c7b2d">

Have others encountered similar issues?

Thanks.

**Comment by shawnlwy85** [comment]
my vps cpu 2 core running 30-40%, but harddisk ipos reach 2000+ cause server hang.

facing high disk usage after deploy to vps server,
and cause 20min-30min website unavailable.

any one face this issue?

------------------------------------ solved
just make sure the server /  vps have minimum 1GB above RAM

**Comment by sean-escaped** [comment]
I have same error, CPU suddenly over 100% , after that Mem use rise over 100% 
Im using Next 13.4

**Comment by ratz6** [comment]
any update here ?

**Reply by shawnlwy85** [reply]
just make sure the server /  vps have minimum 1GB above RAM

for my case
